src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/Select.js

```diff
@@ -118,6 +189,75 @@ class Select<T> extends React.Component<Props<T>> {
 
     handleOptionListClose = this.closeOptionList;
 
+    @computed get highestButtonIndex() {
+        const max = Math.max(...Array.from(this.buttonRefsByIndex.keys()));
+        return max;
+    }
+
+    @action requestFocus = (elementIndex: number) => {
+        if (!this.buttonRefsByIndex.has(elementIndex)) {
+            return;
+        }
+
+        this.focusedElementIndex = elementIndex;
+        const ref = this.buttonRefsByIndex.get(elementIndex);
+
+        if (ref) {
+            ref.focus();
+        }
+    };
+
+    handleRequestFocus = (elementIndex: number) => () => {
+        this.requestFocus(elementIndex);
+    };
+
+    @action handleKeyDown = (event: KeyboardEvent) => {
+        if (!this.open) {
+            return;
+        }
+
+        if (event.key === 'Escape') {
+            event.preventDefault();
+            this.closeOptionList();
+            this.clearSearchText();
+
+            return;
+        }
+
+        let focusedElementIndex = this.focusedElementIndex;
+
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+            this.clearSearchText();
+
+            focusedElementIndex = Math.max(
+                ...this.availableButtonIndices.filter((i) => i < this.focusedElementIndex)
```

Comment:
will this work if you are already at the top of the list? 🤔

Reply:
yes, because the `requestFocus()` checks, if the `buttonRefsByIndex` has that key